providers/discord/src/airflow/providers/discord/hooks/discord_webhook.py

```diff
@@ -70,21 +115,28 @@ def build_discord_payload(
                         (max 2000 characters)
         :param username: Override the default username of the webhook
         :param avatar_url: Override the default avatar of the webhook
+        :param embed: Discord embed object.
         :return: Discord payload (str) to send
         """
-        if len(message) > 2000:
-            raise ValueError("Discord message length must be 2000 or fewer characters.")
-        payload: dict[str, Any] = {
+        payload = {
             "content": message,
             "tts": tts,
         }
+        if len(message) > 2000:
+            raise ValueError("Discord message length must be 2000 or fewer characters.")
```

Comment:
Why do we move it here? Originally we put it there as we want it to fail early

Reply:
I'm sorry, that's my mistake. Changed.